group-coordinator/src/main/java/org/apache/kafka/coordinator/group/consumer/CurrentAssignmentBuilder.java

```diff
@@ -34,48 +33,68 @@
  * consumer group protocol. Given the current state of a member and a desired or target
  * assignment state, the state machine takes the necessary steps to converge them.
  *
- * The member state has the following properties:
- * - Current Epoch:
- *   The current epoch of the member.
+ * State Machine:
+ * - STABLE:
+ *   The member is fully reconciled to the desired target assignment.
  *
- * - Next Epoch:
- *   The desired epoch of the member. It corresponds to the epoch of the target/desired assignment.
- *   The member transitions to this epoch when it has revoked the partitions that it does not own
- *   or if it does not have to revoke any.
+ *   Valid Transitions:
+ *   - STABLE -> STABLE
+ *     When a new target assignment is installed but the assignment of the member
```

Comment:
Just for my understanding, when we say new target assignment installed do we mean the assignment reconciliation on the client aka assignment installed on the client?

Reply:
I think this is the case where we get a new assignment server side, but nothing changed for the member?